index-template.html

```diff
@@ -5,121 +5,58 @@
+{
+    "vizId": "123456",
+    "title": "What is my title",
+    "description": "This is a description",
+    "dataSources": [
+        {
+            "uid": "SomeRandomValue",
+            "type": "cob-arcgis",
+            "icon": "https://patterns.boston.gov/images/global/icons/mapping/waypoint-charles-blue.svg",
+            "clusterIcons": true,
+            "polygonStyle": "default",
+            "popover": "<p>I am a popover</p>",
+            "attributes": {
+                "service": "https://services.arcgis.com/sFnw0xNflSi8J0uh/ArcGIS/rest/services/food_trucks_schedule/FeatureServer",
+                "layer": 0
+            },
+            "legendLabel": "legendary"
         }
-      }
-      $(document).ready(function () {
-        var datatable, apiUrl = ''
-        document.addEventListener('changetable', function (event) {
-          var columnNames = []
-
-          event.detail.columns.forEach(function(item) {
-            columnNames.push(item.name)
-          })
-
-          apiUrl = event.detail.url + '?f=json&returnGeometry=false&outFields=' + columnNames.join(',')
-          datatable = setupTable(event.detail.columns, apiUrl)
-        })
-        document.addEventListener('myvizid.filter', function (event) {
-          $('#fevt').html('<code>' + JSON.stringify(event.detail) + '</code>')
-        })
-        document.addEventListener('changetable', function (event) {
-          console.log('ctevt')
-          $('#ctevt').html('<code>' + JSON.stringify(event.detail) + '</code>')
-        })
-        document.addEventListener('setfilter', function (event) {
-          $('#cfevt').html('<code>' + JSON.stringify(event.detail) + '</code>')
-        })
-        document.addEventListener('myvizid.filter', function (event) {
-          var queryParams = []
-          for (var field in event.detail) {
-            if (event.detail[field]) {
-              if (Array.isArray(event.detail[field])) {
-                queryParams.push(field + ' IN (\'' + event.detail[field].join('\',\'') + '\')')
-              } else {
-                queryParams.push(field + '=\'' + event.detail[field] + '\'')
-              }
-            }
-          }
-          if (queryParams.length === 0) {
-            queryParams = ['1=1']
-          }
-          var searchUrl = apiUrl + '&where=' + encodeURIComponent(queryParams.join(' AND '))
-          datatable.ajax.url(searchUrl).load()
-        })
-      })
+    ],
+    "maps": []
+}
+          </textarea>
+        </div>
+      </li>
+    </ul>
+    <script type="text/javascript">
+    var vizconfiginput = document.getElementById('vizconfig')
+    var vizconfigoutput = document.getElementById('configoutput')
+    vizconfiginput.addEventListener('input', function (e) {
+      let newhtml = JSON.stringify(JSON.parse(vizconfiginput.value), null, '    ')
+      newhtml = newhtml.replace(/</g, '&lt;').replace(/>/g, '&gt;')
+      vizconfigoutput.innerHTML = newhtml
```

Comment:
Actually, I don't think I can in this case. This is updating the `<pre>` content. Since there is raw HTML, the `<` and `>` need to be properly escaped or they will be interpreted as tags.

Reply:
`innerText` automatically escapes.
<img width="273" alt="screen shot 2018-03-01 at 8 51 11 am" src="https://user-images.githubusercontent.com/260837/36847904-b538595c-1d2d-11e8-804d-b3bf334399a1.png">

Reply:
Ah! I didn't read the `innerText`. I read it as `innerHTML`. That makes sense now.